### ISSUE TITLE
Cop idea: redudant self assignment

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8451]
Since methods like `Array#concat` returns `self`:

```ruby
# bad
args = args.concat(ary)
self.foo = foo.concat(ary)

# good
args.concat(foo) # or
args += foo
foo.concat(ary) # or
self.foo += ary
```

Noticed [in our code](https://github.com/rubocop-hq/rubocop/blob/4943d5005b44c61973910b77adbb5fa42209bbfd/lib/rubocop/cop/mixin/check_line_breakable.rb#L174)

**Comment by marcandre** [comment]
Note: I didn't check if we already have a similar cop
